src/playbacks/html5_video/html5_video.js

```diff
@@ -140,12 +140,19 @@ export default class HTML5Video extends Playback {
     this._handleBufferingEvents()
     this.trigger(Events.PLAYBACK_LOADEDMETADATA, {duration: e.target.duration, data: e})
     this._updateSettings()
+    this._checkHighDefinition(e)
     const autoSeekFromUrl = typeof(this._options.autoSeekFromUrl) === 'undefined' || this._options.autoSeekFromUrl
     if (this.getPlaybackType() !== Playback.LIVE && autoSeekFromUrl) {
       this._checkInitialSeek()
     }
   }
 
+  _checkHighDefinition(e) {
+    const currentHeight = e.target.videoHeight
```

Comment:
I would argue that other playbacks would need to do their own different logic because we already have different approaches in hls and html5_video. In hls js HD check happens on level switch, in html5_video it can only happen when new source loads. So by moving it to timeUpdate we're increasing number of checks that can never really change in the context of this playback.